### PR TITLE
Update API documentation and switch to `@properties` for some `Descriptor` attributes.  

### PR DESCRIPTION
**Pull request by andrewdnolan**
This PR:

1. Improves the documentation, switch to numpy style docstrings. The primary improvements are made for the `Descriptor` class, which has improved formatting and more detailed information (for both the class itself and it's attributes). 

2. I've switched the `projection` and `latlon` attributes to be properties. I did this in order to move some logic outside of the method construct to abstract things a little. Most importantly the coordinate arrays (from the minimal dataset) transformation has been moved to the `Descriptor.projection` setter method so that reprojection to a new map projection after Descriptor initialization is now possible, although not really advised. These abstraction should also help set the groundwork for planar-periodic mesh support.

### CODE REVIEW

File: mosaic/descriptor.py [hunk 1]
```diff
@@ -17,60 +20,119 @@
                        "edgesOnVertex"]
 
 
-class Descriptor:
+def attr_to_bool(attr: str):
+    """Format attribute strings and return a boolean value
     """
```
Comment (xylar) [inline mosaic/descriptor.py:25]: I'd typically format a single-line docstring like this:
```suggestion
    """ Format attribute strings and return a boolean value """
```

File: mosaic/descriptor.py [hunk 2]
```diff
@@ -105,63 +167,142 @@ def fix_outofbounds_indices(ds, array_name):
...
+            # rename the coordinate arrays to be named x.../y... irrespective
+            # of whether spherical or Cartesian coordinates are used
             minimal_ds = minimal_ds.rename(renaming_dict)
 
         return minimal_ds
 
+    @property
+    def projection(self) -> CRS:
+        """The target projection for plotting.
+        """
```
Comment (xylar) [inline mosaic/descriptor.py:190]:
```suggestion
        """ The target projection for plotting. """
```

File: mosaic/descriptor.py [hunk 3]
```diff
@@ -105,63 +167,142 @@ def fix_outofbounds_indices(ds, array_name):
...
+                if attr in self.__dict__:
+                    del self.__dict__[attr]
+
+        self._projection = projection
+
+    @property
+    def latlon(self) -> bool:
+        """Boolean whether the lat/lon coordinate arrays should be used for
+        patch construction.
+        """
```
Comment (xylar) [inline mosaic/descriptor.py:221]:
```suggestion
        """
        Boolean whether the lat/lon coordinate arrays should be used for
        patch construction.
        """
```

File: mosaic/descriptor.py [hunk 4]
```diff
@@ -105,63 +167,142 @@ def fix_outofbounds_indices(ds, array_name):
...
+        return self._latlon
+
+    @latlon.setter
+    def latlon(self, value) -> None:
+        """TODO: check that the passed value is consistent with transform
+        """
```
Comment (xylar) [inline mosaic/descriptor.py:227]:
```suggestion
        """ TODO: check that the passed value is consistent with transform """
```

**Review by xylar — APPROVED** [review]
@andrewdnolan, these changes look great to me!  I have some picky suggestions about single-line docstrings below.

I built the docs locally and looked through them.  Everything looked as expected -- very clean!

**Comment by andrewdnolan** [comment]
# Testing 

Here's a demonstration of the reprojection capabilities enable by the `Descriptor.projection` setter method: 
```python 
import cartopy.crs as ccrs
import mosaic
import matplotlib.pyplot as plt
import xarray as xr

def make_figure(ds, descriptor):

    fig, ax = plt.subplots(
        figsize=(9,7), constrained_layout=True, subplot_kw=dict(projection=descriptor.projection)
    )

    collection = mosaic.polypcolor(ax, descriptor, ds.indexToCellID, antialiaseds=True)

    ax.coastlines()
    fig.colorbar(collection, fraction=0.1, shrink=0.5, label="Cell Index");

    return fig, ax

ds = mosaic.datasets.open_dataset("QU.240km")

projection_one = ccrs.InterruptedGoodeHomolosine()
projection_two = ccrs.LambertConformal()
transform = ccrs.PlateCarree()

descriptor = mosaic.Descriptor(ds, projection_one, transform)

# first figure with original projection 
fig_one, ax_one = make_figure(ds, descriptor)

# switch the projection of the descriptor 
descriptor.projection = projection_two

# second figure with new projection
fig_two, ax_two = make_figure(ds, descriptor)

plt.show()
```

## Figure One: 
Screenshot 2024-12-19 at 11 59 54 AM

## Figure Two: 
Screenshot 2024-12-19 at 12 00 27 PM

**Pull request merged**